onnxruntime/test/platform/ios/ios_package_test/ios_package_testTests/ios_package_test_c_api.m

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+//
+//  ios_package_test_c_api.m
+//  ios_package_testTests
+//
+//  This file hosts the tests of ORT C API, for tests of ORT C++ API, please see ios_package_test_cpp_api.mm
+//
+
+#import <XCTest/XCTest.h>
+#include <math.h>
+#include <onnxruntime/onnxruntime_c_api.h>
+
+#define ASSERT_ON_ERROR(expr)                                                            \
```

Comment:
>ASSERT_ON_ERROR [](http://example.com/codeflow?start=8&length=15)

do you want to stop if the assertion fails?
I found out the hard way that XCTAssert* calls will not stop the rest of the function from running by default.
setting continueAfterFailure = NO (e.g., in setUp) will make an assertion failure stop the test.
[https://developer.apple.com/documentation/xctest/xctestcase/1496260-continueafterfailure?language=objc](https://developer.apple.com/documentation/xctest/xctestcase/1496260-continueafterfailure?language=objc)

Reply:
if it does stop on failure, then resources could leak. but if not, we might use uninitialized resources

Reply:
I guess I will keep it this way, the project has only 2 test cases, I guess use uninitialized resources -> crash is fine for now